chasm/lib/scheduler/scheduler.go

```diff
@@ -51,9 +53,18 @@ type Scheduler struct {
 const (
 	// How many recent actions to keep on the Info.RecentActions list.
 	recentActionCount = 10
+
+	// Item limit per spec field on the ScheduleInfo memo.
+	listInfoSpecFieldLimit = 10
+
+	// Field in which the schedule's memo is stored.
+	visibilityMemoFieldInfo = "ScheduleInfo"
 )
 
-var ErrConflictTokenMismatch = serviceerror.NewFailedPrecondition("mismatched conflict token")
+var (
+	ErrConflictTokenMismatch = errors.New("mismatched conflict token")
+	ErrUnprocessable         = errors.New("unprocessable schedule")
```

Comment:
You need to make sure that any errors you return as part of API responses have the correct status code attached and that task failures are marked as unprocessable if they cannot be retried.

I didn't go over all of the locations where these errors are returned but you should do a scan and verify that you are returning the correct status codes everywhere.